### PR TITLE
Add rw cli tests to e2e

### PR DESCRIPTION
**Pull request by dac09**
### What?
Adds tests for running a few commands using the rw cli
- `rw test api`
- `rw build --no-prerender`
- `rw prerender`

### Headsup
The cli command tests rely on the tutorial tests running first, which why we have the `01-tutorial` and `02-cli` names

### Note ℹ️
I changes the updated run-e2e script to take an optional `--prefer-offline` argument, as this was causing issues for me locally

### CODE REVIEW

File: tasks/run-e2e [hunk 1]
```diff
@@ -136,6 +140,11 @@ const args = yargs
   .option('copy-framework', { default: true, type: 'boolean', alias: 'copy' })
   .option('js-project', { default: true, type: 'boolean', alias: 'js' })
   .option('auto-start', { default: true, type: 'boolean', alias: 'start' })
+  .option('prefer-offline', {
```
Comment (dac09) [inline tasks/run-e2e:143]: @peterp made this optional

Comment (peterp) [reply tasks/run-e2e:157]: I think if it's causing problem let's just remove it. It's possible to set this via `.yarnrc` as well.

Comment (dac09) [reply tasks/run-e2e:157]: I think the problem was actually the date, but I think while having prefer-offline is useful, it's better to not use it by default (to keep the installation more accurate to the user's flow).

File: tasks/e2e/cypress/integration/02-cli/02-cli-commands.spec.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
+/* eslint-disable no-undef, camelcase */
+/// <reference types="cypress" />
+
+const BASE_DIR = Cypress.env('RW_PATH')
+
+describe('Check Redwood cli commands against tutorial', () => {
+  before(() => {
+    Cypress.config('record', false)
```
Comment (peterp) [inline tasks/e2e/cypress/integration/02-cli/02-cli-commands.spec.js:8]: Could you add a comment on why this is required?

File: tasks/run-e2e [hunk 3]
```diff
@@ -58,15 +58,15 @@ const copyFrameworkPackages = () => {
   try {
     console.log(
       'cp -r',
-      path.join(REDWOODJS_FRAMEWORK_PATH, 'packages'),
-      path.join(REDWOOD_PROJECT_DIRECTORY, 'packages')
+      `"${path.join(REDWOODJS_FRAMEWORK_PATH, 'packages')}"`,
+      `"${path.join(REDWOOD_PROJECT_DIRECTORY, 'packages')}"`
```
Comment (peterp) [inline tasks/run-e2e:62]: Spaces :shakes-fist!

File: tasks/run-e2e (hunk 1 above)
Comment (peterp) [reply tasks/run-e2e:157]: Yeah, let's take it out completely. It's mostly useless if it's not on by default, and can be overwritten by a user's own configuration.

The problem was the date?

File: tasks/run-e2e [hunk 4]
```diff
@@ -144,12 +167,22 @@ const args = yargs
...
 
 console.log()
 console.log('-'.repeat(80))
 console.log()
-makeDirectory(REDWOOD_PROJECT_DIRECTORY)
+if (args._?.[0]) {
+  console.log('Using supplied directory \n', REDWOOD_PROJECT_DIRECTORY)
+} else {
+  makeDirectory(REDWOOD_PROJECT_DIRECTORY)
+}
```
Comment (peterp) [inline tasks/run-e2e:185]: The intention here is that a user can also supply a path. If the path doesn't exist it'll be created, but if it does then nothing happens.

Super helpful when trying to debug something:
`rm -rf /tmp/rwjs-test; ./tasks/run-e2e; cat /tmp/rwjs-test/web/src/App.tsx`

File: tasks/run-e2e (hunk 1 above)
Comment (dac09) [reply tasks/run-e2e:157]: Having `:` in the folder name :) https://github.com/redwoodjs/redwood/pull/1932/files#diff-e4d8d48827bb7da131ab93737cf7a718801e74e8c86b22e011f27dd4f948b1edR174

File: tasks/run-e2e [hunk 5]
```diff
@@ -162,12 +195,14 @@ const tasks = [
...
   autoStart && runDevServerInBackground,
+  autoStart && installTaskDeps,
   autoStart && runCypress,
 ].filter(Boolean)
 
 for (const task of tasks) {
   console.log()
-  task()
+  // Also pass all args to each task
+  task(args)
```
Comment (peterp) [inline tasks/run-e2e:205]:
```suggestion
  // Also pass all args to each task
  task()
```
Also remove when you remove the `preferOffline` stuff.

**Review by peterp — CHANGES_REQUESTED** [review]

File: tasks/run-e2e (hunk 4 above)
Comment (dac09) [reply tasks/run-e2e:172]: Ah gotcha. Will remove the block.

Comment (dac09) [reply tasks/run-e2e]: Done

**Review by peterp — APPROVED** [review]

**Pull request merged**